src/java.base/share/classes/java/lang/StringLatin1.java

```diff
@@ -830,7 +830,7 @@ static Stream<String> lines(byte[] value) {
     static void putCharsAt(byte[] val, int index, int c1, int c2, int c3, int c4) {
         assert index >= 0 && index + 3 < length(val) : "Trusted caller missed bounds check";
         // Don't use the putChar method, Its instrinsic will cause C2 unable to combining values into larger stores.
-        long address  = Unsafe.ARRAY_BYTE_BASE_OFFSET + index;
+        long address  = (long) Unsafe.ARRAY_BYTE_BASE_OFFSET + index;
         UNSAFE.putByte(val, address    , (byte)(c1));
         UNSAFE.putByte(val, address + 1, (byte)(c2));
         UNSAFE.putByte(val, address + 2, (byte)(c3));
```

Comment:
While you are here, I wonder if this should be renamed to offset to make it clearer.